tesseract_kinematics/include/tesseract_kinematics/core/inverse_kinematics.h

```diff
@@ -131,6 +131,12 @@ class InverseKinematics
    */
   virtual void setLimits(tesseract_common::KinematicLimits limits) = 0;
 
+  /**
+   * @brief Get vector indicating which joints are capable of producing redundant solutions
+   * @return A vector of joint indicies
+   */
+  virtual const std::vector<Eigen::Index>& getRedundancyCapableJointIndices() const = 0;
```

Comment:
Does it make sense for a pure virtual function to return a const reference? It's one thing for a class to return a reference of a member that will always exist, but it's another thing to require the implementation of this interface to keep the `std::vector<Eigen::Index>&` in scope for as long as it will be used (which the implementation can't know). I would suggest just returning the object instead of a referenece. It will probably be small and this method probably won't be called frequently, so we won't be paying a noticeable performance penalty by doing this
```suggestion
  virtual const std::vector<Eigen::Index> getRedundancyCapableJointIndices() const = 0;
```

Reply:
I do not see an issue with a pure virtual returning a const reference in this case because the information should be known at construction. This just forces that they are stored as member variables so they can be returned as a const reference.

Reply:
> the information should be known at construction

For the majority of cases, yes. My issue with this is mainly conceptual. The restrictions an interface imposes should be as minimal as possible to keep the interface flexible. It's possible that a use-case might want to compute this dynamically so redundant solutions don't get generated for a particular joint. If you restrict the output type to being a reference, it makes things more difficult for that case. There's almost no performance penalty for returning the actual object, so it seems like we're making the interface more restrictive for no reason

Reply:
> It's possible that a use-case might want to compute this dynamically so redundant solutions don't get generated for a particular joint.

I do see your point related to flexibility versus performance on the interface. In this case, I selected performance because I don't see how it is possible given the current interface where dynamically calculating the redundant indices would make sense because the function does not take any inputs?

Reply:
I made the change to the interface.